server/tasks/process_file.py

```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
     asyncio.run(agent.environment.finalize())
 
 
-@app.task(name="process_file_streaming", track_started=True, bind=True)
-def process_file_streaming(self, serialized_agent: bytes):
-    # Load the agent
-    agent = pickle.loads(serialized_agent)
+@app.task(
+    name="streaming_parent_task", track_started=True, bind=True, serializer="pickle"
+)
+def streaming_parent_task(
```

Comment:
nit: can we name this something like `streaming_coordinator_task` or something, to make the responsibility of the task more clear?

Reply:
Hmm this is pretty clear to me IMO haha

Reply:
not the hill I'll die on, but I could imagine a new dev having questions like "why is there a parent job?" and "what is this 'parent job' actually doing?".
I also like that in our diagrams we can have a more meaningful name for these.
That said, "coordinator" isn't too much more descriptive than "parent job", so if this isn't convincing that's fine.

Reply:
Id rather leave a comment on what its doing, than change the name to what I think isnt any more obvious really so Ill do that